subsys/dfu/boot/mcuboot.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2016-2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "fota/mcuboot"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_FOTA_LEVEL
+#include <logging/sys_log.h>
+
+#include <stddef.h>
+#include <errno.h>
+#include <string.h>
+#include <flash.h>
+#include <zephyr.h>
+#include <init.h>
+
+#include "boot/mcuboot.h"
+#include "boot/mcuboot_constrains.h"
+
+/*
+ * Helpers for image trailer, as defined by mcuboot.
+ */
+
+#define TRAILER_IMAGE_MAGIC_SIZE (4 * sizeof(u32_t))
+#define TRAILER_SWAP_STATUS_SIZE (128 * FLASH_MIN_WRITE_SIZE * 3)
```

Comment:
This looks like it might be out of date with the current mcuboot code. We've gotten rid of the need for update software to need to know the status size. The image magic has been moved to the end, and is the only thing that needs to be written (possibly with the image_ok already set).